haystack/__init__.py

```diff
@@ -20,7 +20,8 @@ class Finder:
 
     def __init__(self, reader, retriever):
         self.retriever = retriever
-        self.retriever.fit()
```

Comment:
We actually already call fit() when initalizing the TFIDF retriever (https://github.com/deepset-ai/haystack/blob/elasticsearch_datastore/haystack/retriever/tfidf.py#L36) 
I think we only need one of them and then I would argue to get rid of it here in the Finder()